pkg/resource/validations/kms_arn_regex_validation.go

```diff
@@ -10,10 +10,6 @@ var KmsArnRE = regexp.MustCompile(
 )
 
 func ValidateKMSKeyARN(kmsKeyARN *string) error {
-	if kmsKeyARN == nil || *kmsKeyARN == "" {
-		return fmt.Errorf("expected a non empty value for kms-key-arn")
```

Comment:
Just return nil here, as we need some check here. 

If we pass in a nil string we will get a nil pointer when dereferencing below. 

If nil or empty just return nil